toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -0,0 +1,163 @@
+		partitionFlags     []string
+		partitionTableType PartitionTableType
+		diskInfo           map[string]int
+	)
+
+	// Check whether the config already contains partition schema
+	if len(config.Disks) > 0 && len(config.Disks[0].Partitions) > 0 {
+		logger.Log.Infof("Partition scheme already exists")
+		return
+	}
+
+	// Check whether the incoming config is for initrd
+	if config.SystemConfigs[0].Name == "ISO initrd" {
+		return
+	}
+
+	file, err := os.Open(partitionFile)
+	if err != nil {
+		logger.Log.Errorf("Failed to open file (%s)", partitionFile)
+		return
+	}
+	defer file.Close()
+
+	// Check if the file is empty
+	_, err = file.Stat()
+	if err != nil {
+		logger.Log.Errorf("Empty partition file (%s)", partitionFile)
+		return
+	}
+
+	defaultDiskIndex := 0
+
+	scanner := bufio.NewScanner(file)
+
+	// Create a mapping between the disk path and the index of the disk in the
+	// Disk array so that when we parse through the partition commands, we can
+	// determine which disk the parition instruction is targeted
+	diskInfo = make(map[string]int)
+	partitionTableType = PartitionTableTypeGpt
+
+	for scanner.Scan() {
+		parseCmd = scanner.Text()
+		//fmt.Printf("Parse Commands: %s\n", parseCmd)
+
+		// Find disk information
+		if strings.Contains(parseCmd, "--ondisk") {
+			partitionFlags = strings.Split(parseCmd, " ")
+			curDiskInfo := partitionFlags[len(partitionFlags)-1]
+			curDiskInfo = curDiskInfo[9:len(curDiskInfo)]
+
+			// Check whether this disk has already been parsed or not
+			curIdx, ok := diskInfo[curDiskInfo]
+			if !ok {
+				// Create new disk struct and append it into the Disk array
+				newDisk := Disk{}
+				newDisk.PartitionTableType = partitionTableType
+
+				// Set TargetDisk and TargetDiskType for unattended installation
+				newDisk.TargetDisk.Type = "path"
+				newDisk.TargetDisk.Value = curDiskInfo
+
+				diskInfo[curDiskInfo] = defaultDiskIndex
+				curIdx = defaultDiskIndex
+				defaultDiskIndex++
+
+				config.Disks = append(config.Disks, newDisk)
+			}
+
+			// Create new Partition and PartionSetting
+			partition := new(Partition)
+			partitionSetting := new(PartitionSetting)
+			partitionSetting.MountIdentifier = MountIdentifierDefault
+
+			// Find Mount Point
+			partitionSetting.MountPoint = partitionFlags[1]
+			if partitionFlags[1] == "biosboot" {
+				partitionSetting.MountPoint = ""
+				partitionSetting.ID = "boot"
+				partition.ID = "boot"
+				partition.Flags = append(partition.Flags, PartitionFlagBiosGrub)
+				config.Disks[curIdx].PartitionTableType = PartitionTableTypeGpt
+			} else if partitionFlags[1] == "/" {
+				partitionSetting.ID = "rootfs"
+				partition.ID = "rootfs"
+			} else {
+				partitionSetting.ID = partitionFlags[1]
+				partition.ID = partitionFlags[1]
+			}
+
+			// Loop through partition flags to fetch fstype, size etc.
+			for _, partOpt := range partitionFlags {
+
+				// Find fstype
+				if strings.Contains(partOpt, "--fstype") {
+					fstype := partOpt[9:len(partOpt)]
```

Comment:
This approach does not take into the account `--flagname=flagvalue`, which seems to be legal kickstart grammar: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/installation_guide/s1-kickstart2-options

I suggest changing the approach here altogether; instead of searching strings multiple times for different expected symbols, split the input by whitespace and match pairs `--name` with values.

Reply:
I wonder if we can use the kingpin parser we use to parse arguments to our go tools somehow? I'm not sure exactly how it hooks up.

Reply:
Has refactored the implementation